src/main/java/com/opinta/service/DiscountPerCounterpartyServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+        DiscountPerCounterparty discountPerCounterparty = discountPerCounterpartyDao.getByUuid(uuid);
+        if (discountPerCounterparty == null) {
+            log.error(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, uuid));
+            throw new IncorrectInputDataException(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, uuid));
+        }
+
+        userService.authorizeForAction(discountPerCounterparty, user);
+
+        return discountPerCounterpartyDao.getByUuid(uuid);
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterparty saveEntity(DiscountPerCounterparty discountPerCounterparty, User user)
+            throws AuthException, IncorrectInputDataException, PerformProcessFailedException {
+        Counterparty counterparty = counterpartyService.getEntityByUuid(
+                discountPerCounterparty.getCounterparty().getUuid(), user);
+        Discount discount = discountService.getEntityByUuid(discountPerCounterparty.getDiscount().getUuid());
+
+        discountPerCounterparty.setCounterparty(counterparty);
+        discountPerCounterparty.setDiscount(discount);
+        validateDiscountTime(discountPerCounterparty);
+        log.info(LogMessageUtil.saveLogEndpoint(DiscountPerCounterparty.class, discountPerCounterparty));
+        return discountPerCounterpartyDao.save(discountPerCounterparty);
+    }
+    
+    @Override
+    public DiscountPerCounterparty updateEntity(DiscountPerCounterparty discountPerCounterparty, User user)
+            throws AuthException, IncorrectInputDataException, PerformProcessFailedException {
+        userService.authorizeForAction(discountPerCounterparty, user);
+        validateDiscountTime(discountPerCounterparty);
+        discountPerCounterpartyDao.update(discountPerCounterparty);
+        return discountPerCounterparty;
+    }
+    
+    
+    @Override
+    @Transactional
+    public List<DiscountPerCounterpartyDto> getAll(User user) {
+        return discountPerCounterpartyMapper.toDto(getAllEntities(user));
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterpartyDto getByUuid(UUID uuid, User user) throws IncorrectInputDataException,
+            AuthException {
+        return discountPerCounterpartyMapper.toDto(getEntityByUuid(uuid, user));
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterpartyDto save(DiscountPerCounterpartyDto discountPerCounterpartyDto, User user)
+            throws IncorrectInputDataException, AuthException, PerformProcessFailedException {
+        DiscountPerCounterparty discountPerCounterparty = discountPerCounterpartyMapper
+                .toEntity(discountPerCounterpartyDto);
+        discountPerCounterparty.setDiscount(discountService.getEntityByUuid(discountPerCounterpartyDto.getDiscountUuid()));
+        validateDiscountTime(discountPerCounterparty);
+        return discountPerCounterpartyMapper.toDto(saveEntity(discountPerCounterparty, user));
```

Comment:
No need to validate and do other stuff here.
Just use mapper and pass data to the method saveEntity(...)